src/sentry/integrations/opsgenie/integration.py

```diff
@@ -120,7 +121,46 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:  # type:ignore
 
 
 class OpsgenieIntegration(IntegrationInstallation):
-    pass
+    def get_client(self) -> Any:
+        org_integration_id = self.org_integration.id if self.org_integration else None
+        return OpsgenieClient(integration=self.model, org_integration_id=org_integration_id)
+
+    def get_organization_config(self) -> Sequence[Any]:
+        fields = [
+            {
+                "name": "team_table",
+                "type": "table",
+                "label": "Opsgenie teams with the Sentry integration enabled",
+                "help": "If teams need to be updated, deleted, or added manually please do so here. Alert rules will need to be individually updated for any additions or deletions of teams.",
+                "addButtonText": "",
+                "columnLabels": {"team": "Team", "integration_key": "Integration Key"},
+                "columnKeys": ["team", "integration_key"],
+                "confirmDeleteMessage": "Any alert rules associated with this team will stop working. The rules will still exist but will show a `removed` team.",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        client = self.get_client()
+        # get the team ID/test the API key for a newly added row
+        teams = data["team_table"]
+        unsaved_teams = [team for team in teams if team["id"] == ""]
+        for team in unsaved_teams:
+            try:
+                resp = client.get_team_id(
+                    integration_key=team["integration_key"], team_name=team["team"]
+                )
+                team["id"] = resp["data"]["id"]
+            except ApiError:
+                raise ValidationError(
+                    {"api_key": ["Could not save: invalid team name or integration key."]}
```

Comment:
nit: I think this can be a regular string instead of a list?
```suggestion
                    {"api_key": "Could not save due to invalid team name or integration key."}
```

Reply:
It actually has to be a list—I just checked. Maybe due to something in the frontend code?